src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -74,7 +80,14 @@ internal FeedAndEntryMaterializerAdapter(ODataFormat odataFormat, ODataReaderWra
             this.currentEntry = null;
             this.currentFeed = null;
             this.feedEntries = null;
+            this.currentDeltaFeed = null;
             this.materializerContext = materializerContext;
+
+            if (isDeltaPayload)
+            {
+                // If we have a delta payload, we need the stack data structure to keep track of the read items.
+                this.items = new Stack<IMaterializerState>();

```

Comment:
I think a name that reflects this use would be better than `this.items`.

Reply:
changes this to materializerStateItems